doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+<nb>user overwritten user action could be a log out, leaving some tuning mode,
+forcing an inference engine update...</nb>
+<why>Even if a parameter is managed by the inference engine, it often is useful
+(test, debugging) to overwrite its value temporally.</why>
+
+A PF tuning capability **MAY** be disabled in a context where no tuning is needed.
+<why>The reference implementation does so (phone end users can not change the
+tuning).</why>
+
+# Command line interface
+
+The PF **MAY** offer a command line interface that binds to its IPC.
+<why>to have a reference way to interact with a PF without implementing its IPC
+protocol.</why>
+
+This command line interface **SHOULD** support all tuning and introspection
+ability.
+<why>In order to be used in scripting and live tuning/debugging on an embedded
+system.</why>
+
+This command line interface **MAY** offer argument auto completion.
+<why>Is more user friendly.</why>
+
+# Bindings
+The PF **SHOULD** expose its API in C.
+<why>The PF aims to be a hardware abstraction thus middle ware which is often
+written in C or a language compatible with C. Virtually all programing language
+support C Foreign Procedure Call, having a C API ease integration whichever the
+host language is.</why>
+
+The PF **MAY** expose its API to multiple programing language.
+<why>The reference implementation has python bindings.</why>
+
+# Performance
+
+The reference Parameter Framework implementation is mainly intended for use
+in consumer electronics such as smartphones and tablets. Such platforms are
+often referred to as "embedded" platforms but their capacity today is so huge in
+terms of both computing and memory that they can be considered as small personal
+computers.
+
+Moreover, since one of the Parameter Framework's primary feature is to implement
+storage of a) a hardware description and b) settings, its memory footprint
+largely depends on how many such items are stored.
+
+For those reasons, there are no performance requirements imposed on the
+architecture. Performance considerations are left to the implementation of the
+Parameter Framework and/or the client and/or the build chain.
+
+# Next
+<ko>
+The following requirements are not implemented in the reference implementation
+and are to be considered draft.
+</ko>
+
+PF **MAY** support at least:
+
+ - Linux (and Android)
+ - Windows
+ - Mac OSX
+
+<why>As the reference PF implementation leaves its original Android environment,
+needs emerge to use it on other platform.</why>
+
+The PF host API **SHOULD** be structured.
+I.e.: the PF, when requested for a list of domains, should return a list of
+structured object, each containing configuration objects, containing their
+values.
+<why>The reference implementation has a string oriented API. E.g/: The list of
+domains is returned as a concatenation of domains name in one big string. This
+leads to hard an hard to use API from C and C++ code.</why>
```

Comment:
You comments are very pertinent and welcomed, although it is true a bit late. Nevertheless better late then never.